src/orquestra/quantum/evolution.py

```diff
@@ -36,20 +37,18 @@ def time_evolution(
     if method != "Trotter":
         raise ValueError(f"Currently the method {method} is not supported.")
 
-    terms: Iterable = list(hamiltonian.get_operators())
-
     return reduce(
         operator.add,
         (
             time_evolution_for_term(term, time / trotter_order)
             for _index_order in range(trotter_order)
-            for term in terms
+            for term in hamiltonian.terms
```

Comment:
praise: This is so much more clear than the way we had it. 😄

Reply:
Thanks 😁